src/ucp/api/ucp.h

```diff
@@ -3573,6 +3622,35 @@ ucs_status_ptr_t ucp_worker_flush_nb(ucp_worker_h worker, unsigned flags,
                                      ucp_send_callback_t cb);
 
 
+/**
+ * @ingroup UCP_WORKER
+ *
+ * @brief Flush outstanding AMO and RMA operations on the @ref ucp_worker_h
+ * "worker"
+ *
+ * This routine flushes all outstanding AMO and RMA communications on the
+ * @ref ucp_worker_h "worker". All the AMO and RMA operations issued on the
+ * @a worker prior to this call are completed both at the origin and at the
+ * target when this call returns.
+ *
+ * @note For description of the differences between @ref ucp_worker_flush_nb
+ * "flush" and @ref ucp_worker_fence "fence" operations please see
+ * @ref ucp_worker_fence "ucp_worker_fence()"
+ *
+ * @param [in] worker    UCP worker.
+ * @param [in] param     Operation parameters, see @ref ucp_request_param_t
```

Comment:
pls document which flags could be specified in param->flags

Reply:
in current implementation flags are not used, will update docs after new functionality added

Reply:
pls add note about operation-specific flags

Reply:
added note

Reply:
@yosefe I think you request was more specific. Do you suggest explicitly list the list of parameters under ucp_request_param_t that are relevant for each one of the operations ? I don't this has been done.

Reply:
for now there is no defined flags used in flush operations

Reply:
yep, for now the "ucp_worker_flush_nb" operation does not have defined flags, it's "Reserved for future use".

Reply:
@hoopoepg - is this true for all the above APIs ? Can I pass NULL as a param if it not used ?

Reply:
flags value is just ignored in API's where it is not used, you may set it to NULL or leave uninitialized

Reply:
and it is true for all other API's

Reply:
Let's be explicit about this. My thoughts 
* It is important to note that in future, it will impact number of branches in the API 
* The ignore option has to be consistent across UCP
* Leaving this undefined sounds like poor choice.

Reply:
updated comment - user should set it to 0